Create a section in the history panel to reset scene activity

ikmckenz opened the pull request:
This resolves the main concern behind #5035 in a way that fits with the existing workflow to manage o history and view history. Tested working locally.

WithoutPants commented on pkg/sqlite/scene.go:
```diff
@@ -1234,6 +1234,23 @@ func (qb *SceneStore) SaveActivity(ctx context.Context, id int, resumeTime *floa
+		return false, err
+	}
+
+	record := goqu.Record{}
+
+	record["resume_time"] = 0.0
+	record["play_duration"] = 0.0
+
+	if err := qb.tableMgr.updateByID(ctx, id, record); err != nil {
+		return false, err
+	}
```
This can be simplified to:
```
if err := qb.tableMgr.updateByID(ctx, id, goqu.Record{
	"resume_time":   0.0,
	"play_duration": 0.0,
}); err != nil {
	return false, err
}```

Or alternatively it can be in the `updateByID` call.

ikmckenz replied:
After moving to boolean flags for resume/duration, the simplification is no longer possible.

WithoutPants reviewed (changes requested):
Thanks for this. For future reference, it's good to include screenshots of UI changes so that reviewers can assess without having to run a build themselves.

![image](https://github.com/user-attachments/assets/bc2d0ef1-adbd-495f-beda-046c30767fbc)

From what I can see, you've added a `Resume Time` heading with a trash icon button. This calls `sceneResetActivity` which sets both the `resume_time` _and_ `play_duration` fields to `0`.

I think this current design is a little confusing for the average user. The meaning of the `Resume Time` heading isn't clear, and there's no obvious indication that clicking the trash button will reset both the resume time _and_ the play duration. 

I think there's probably a use case where the user wants to reset the resume time _without_ resetting the play duration, so these functions should probably be split up. The existing `sceneResetActivity` interface can be kept, but I'd add boolean flags for resetting play duration and/or resume time.

Given that this roughly falls under the `Play History` category, I think these functions are better placed in the Play History dropdown button.

ikmckenz commented:
Thank you for the review! 

As requested I have split the functionality into separate "reset resume" and "reset duration" actions, and moved them to the Play History dropdown button. On the backend I have kept the `sceneResetActivity` but added the booleans to separate the different activities as requested. The only thing is that I didn't simplify the `ResetActivity`  as originally requested, since with the booleans it works better copying the format of the function above it (`SaveActivity`). 

The proposed UI change now looks like this: 
<img width="443" alt="Screenshot 2024-08-27 at 11 55 22 PM" src="https://github.com/user-attachments/assets/470dbcc6-a7c8-4d79-8695-964ba476544f">

WithoutPants reviewed (approved):
Thanks! Just made some minor changes:
- made the `reset_` arguments optional so that clients don't have to pass both
- added Toast popups after success or failure for the reset operations

ikmckenz commented:
Thank you!
